kvdb-rocksdb/src/lib.rs

```diff
@@ -680,6 +681,15 @@ impl Database {
 			HashMap::new()
 		}
 	}
+
+	/// Try to catch up a secondary instance of the database with
+	/// the primary instance by reading as much from the logs as possible
```

Comment:
Would be good to provide some info on when (if ever?) it is impossible for a secondary to catch up.
Also: does this block until the secondary has caught up? Worth mentioning here I think.
Once caught up, is the secondary guaranteed to have all data in the DB up to that time stamp? Does catching up have any implications for the primary, e.g. if the primary is in the middle of committing a write, does the secondary wait until the primary is done?

Reply:
I'm still not entirely certain of the implementation, however I'm reasonably sure of the following:

- Secondary does not acquire any locks that affect the primary
- Secondary only asserts that no lock is held right before reading from the Manifest file. I am pretty sure this guarantees that whatever is read at the timestamp that `try_catch_up_with_primary()` is called is included and applied to the secondary, since a write would require a lock at that time
    - this also means that new writes can occur from the primary while the secondary is reading from the manifest, but the secondary wouldn't get those updates until it calls `try_catch_up_with_primary` again

so, it's not guaranteed that the secondary is caught up with the most recent state as the primary, however if the secondary tries to catch-up right as the primary is writing, the catch-up would fail